src/app.rb

```diff
@@ -6,42 +6,178 @@
 require_relative 'person'
 require_relative 'rental'
 require_relative 'book'
+require 'json'
 
 class App
   def initialize
     @books = []
     @persons = []
     @rentals = []
+
+    data_from_json
   end
 
   def list_all_books
     @books
   end
 
+  def create_a_book(title, author)
+    @books.push(Book.new(title, author))
+  end
+
   def list_all_persons
     @persons
   end
 
   def create_a_teacher(name, age, specialization)
-    @persons.push(Teacher.new(name, age, specialization, parent_permission: true))
+    @persons.push(Teacher.new(age, specialization, name, parent_permission: true))
   end
 
   def create_a_student(name, age, parent_permission)
     @persons.push(Student.new(age, nil, name, parent_permission: parent_permission))
   end
 
-  def create_a_book(title, author)
-    @books.push(Book.new(title, author))
-  end
-
   def create_a_rental(date, book_id, person_id)
-    @rentals.push(Rental.new(date, @books[book_id], @persons[person_id]))
+    @rentals.push(Rental.new(date, @persons[person_id], @books[book_id]))
   end
 
   def list_rental_person(id)
-    @persons.each do |person|
-      return person.rentals if person.id == id
+    person = @persons.find { |p| p.id == id }
+    person&.rentals || []
+  end
+
+  def save_data_to_json
+    save_books_to_json
+    save_people_to_json
+    save_rentals_to_json
+  end
+
+  def data_from_json
+    books_from_json
+    people_from_json
+    rentals_from_json
+  end
+
+  def exit_app
+    save_data_to_json
+    create_empty_json_file('./data/books.json') unless File.exist?('./data/books.json')
+    create_empty_json_file('./data/people.json') unless File.exist?('./data/people.json')
+    create_empty_json_file('./data/rentals.json') unless File.exist?('./data/rentals.json')
+    puts 'Exiting the application.'
+  end
```

Comment:
Great job team🤝

- [ ] As stated in the project requirements, on exit from the student's app, make sure the data is saved (all the data: books, people, and rentals) into JSON files. This is not the case for your project. I am getting this error whenever I exit the app.
![json-error](https://github.com/PabloBona/school_library/assets/63917725/ca23dca5-1051-4426-a9d6-5622cc4fc919)

- [ ] On startup of the app, load all data from files. Handle the case when one or more of the files are missing (the app should work and show no errors).

Reply:
- [ ] @Team great work trying to implement the change requested, however am still getting the same error when I choose option 7 to exit the app, Kindly consider checking how you reference your JSON files.
![image](https://github.com/PabloBona/school_library/assets/61408860/598c85d9-743a-41a6-a3d5-c9860129f332)

Reply:
Hello @danmainah , Thanks for reviewing our code.
I think the problem is that you have not navigated to the `src` directory in your terminal while running the `main.rb` file.
Would you please first go to the `./src` folder by `cd src/` and then run the file again?